tests/unittest_inference.py

```diff
@@ -1732,8 +1734,7 @@ def __init__(self):
         """
         ast = extract_node(code, __name__)
         expr = ast.func.expr
-        with pytest.raises(InferenceError):
-            next(expr.infer())
+        self.assertIs(next(expr.infer()), util.Uninferable)
```

Comment:
This revert the behaviour of this test to what is was before:
https://github.com/PyCQA/astroid/pull/927/files#r610819685

Reply:
Damn the last comment seems to predict the possible InferenceError. I wonder why we merged anyway.

Reply:
I also don't really understand why the behaviour of the test changed because of that change.. `astroid` can be so hard to follow with all the decorators and `context` being passed around..